### PR TITLE
[DEBUG][CI]travis: add coccinelle check for SOF

### PR DESCRIPTION
**Pull request by xiulipan**
Add coccinelle check for SOF part code in Travis CI.
Build coccinelle 1.0.7 from source in Travis CI.

Signed-off-by: Pan Xiuli <xiuli.pan@linux.intel.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,11 +21,17 @@ jobs:
         - scripts/checkpatch.pl --strict --codespell -g $SHA_MAIN..$SHA_PR
     - name: "sparse check"
```
Comment (plbossart) [inline .travis.yml]: It's "Sparse/Coccicheck checks" now.

Comment (xiulipan) [reply .travis.yml]: @plbossart 
How about make it to be a generic name?
`code checks`

Comment (plbossart) [reply .travis.yml]: sure

**Comment by xiulipan** [comment]
@plbossart 
updated

File: .travis.yml [hunk 2]
```diff
@@ -18,13 +19,19 @@ jobs:
...
+        - wget https://github.com/coccinelle/coccinelle/archive/1.0.7.tar.gz
+        - tar -xf 1.0.7.tar.gz
+        - cd coccinelle-1.0.7 && ./autogen && ./configure && make && sudo make install
+        - cd -
         - export ARCH=x86_64
         - make defconfig
         - scripts/kconfig/merge_config.sh .config kconfig/base-defconfig kconfig/sof-defconfig kconfig/hdaudio-codecs-defconfig
         - make modules_prepare
         - make SUBDIRS=sound/soc/sof C=2
+        - make coccicheck MODE=report M=sound/soc/sof
```
Comment (plbossart) [inline .travis.yml:34]: While you are at it you should fix the sparse command and replace SUBDIRS= by M=

Comment (xiulipan) [reply .travis.yml:34]: @plbossart 
Thanks, good point. The log is suggesting to do that.

**Pull request merged**